RecoPixelVertexing/PixelVertexFinding/test/gpuVertexFinder_t.cu

```diff
@@ -0,0 +1,179 @@
+#include<random>
+#include<vector>
+#include<cstdint>
+#include<cmath>
+
+#include "RecoPixelVertexing/PixelVertexFinding/src/gpuClusterTracks.h"
+using namespace  gpuVertexFinder;
+#include <cuda/api_wrappers.h>
+
+
+struct Event {
+  std::vector<float> zvert;
+  std::vector<uint16_t>  itrack;
+  std::vector<float> ztrack;
+  std::vector<float> eztrack;
+  std::vector<uint16_t> ivert;
+};
+
+struct ClusterGenerator {
+
+  explicit ClusterGenerator(float nvert, float ntrack) :
+    rgen(-13.,13), errgen(0.005,0.025), clusGen(nvert), trackGen(ntrack), gauss(0.,1.)
+  {}
+
+  void operator()(Event & ev) {
+
+    int nclus = clusGen(reng);
+    ev.zvert.resize(nclus);
+    ev.itrack.resize(nclus);
+    for (auto & z : ev.zvert) { 
+       z = 3.5f*gauss(reng);
+    }
+
+    ev.ztrack.clear(); 
+    ev.eztrack.clear();
+    ev.ivert.clear();
+    for (int iv=0; iv<nclus; ++iv) {
+      auto nt = trackGen(reng);
+      ev.itrack[nclus] = nt;
+      for (int it=0; it<nt; ++it) {
+       auto err = errgen(reng); // reality is not flat....
+       ev.ztrack.push_back(ev.zvert[iv]+err*gauss(reng));
+       ev.eztrack.push_back(err*err);
+       ev.ivert.push_back(iv);
+      }
+    }
+    // add noise
+    auto nt = 2*trackGen(reng);
+    for (int it=0; it<nt; ++it) {
+      auto err = 0.03f;
+      ev.ztrack.push_back(rgen(reng));
+      ev.eztrack.push_back(err*err);
+      ev.ivert.push_back(9999);
+    }
+
+  }
+
+  std::mt19937 reng;
+  std::uniform_real_distribution<float> rgen;
+  std::uniform_real_distribution<float> errgen;
+  std::poisson_distribution<int> clusGen;
+  std::poisson_distribution<int> trackGen;
+  std::normal_distribution<float> gauss;
+
+
+};
+
+
+#include<iostream>
+
+int main() {
+
+  if (cuda::device::count() == 0) {
+    std::cerr << "No CUDA devices on this system" << "\n";
+    exit(EXIT_FAILURE);
+  }
+
+  auto current_device = cuda::device::current::get();
+
+  auto zt_d = cuda::memory::device::make_unique<float[]>(current_device, 64000);
```

Comment:
can the `64000`s and `256`s be given some name ?

Reply:
this is a test...